box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/BoxSearchView.java

```diff
@@ -202,4 +173,40 @@ public void setOnBoxSearchListener(final OnBoxSearchListener onBoxSearchListener
          */
         void onQueryTextSubmit(String text);
     }
+
+    /**
+     * Provides implementation of {@link OnQueryTextListener} interface. The default
+     * implementation takes care of visibility of the clear button based on the user input.
+     */
+    private class SearchQueryTextListener implements OnQueryTextListener {
```

Comment:
private static
?

Reply:
+1

Reply:
on a second thought, I can't make it static because it is accessing non-static properties of the outer class. Either I let it stay like this or create a property and initialize it using anonymous class:
```
private final SearchQueryTextListener mSearchQueryListener = new SearchQueryTextListener() {
    @Override
    public boolean onQueryTextSubmit(String query) { ... }

   @Override
    public boolean onQueryTextChange(String newText) { ... }
}
```
I would prefer to keep it this way because its just easier to read.